state-sync/state-sync-driver/src/continuous_syncer.rs

```diff
@@ -265,7 +265,7 @@ impl<
 
     /// Returns the highest synced version and epoch in storage
     fn get_highest_synced_version_and_epoch(&self) -> Result<(Version, Epoch), Error> {
-        let highest_synced_version = utils::fetch_latest_synced_version(self.storage.clone())?;
+        let highest_synced_version = utils::fetch_pre_committed_version(self.storage.clone())?;
```

Comment:
do we need both synced version and pre committed version?

Reply:
yes the API can't return data that's after synced version

Reply:
oh.. also the chunk executor on start up needs to know there's a gap